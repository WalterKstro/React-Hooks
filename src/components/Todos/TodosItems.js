import PropTypes from 'prop-types';

const TodosItems = ({todo:{id, task, state},handleToogle,handleDelete}) => {
    return (
        <li 
            key={id} 
            className={`${state && 'text-decoration-line-through text-muted'} fw-bold list-group-item d-flex justify-content-between`}
            onClick={() => handleToogle(id)}>
            {task}
            <button className="btn btn-sm" onClick={() => handleDelete(id)}>🗑️</button>
        </li>
    )
}

TodosItems.propTypes = {
    todo: PropTypes.object.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToogle: PropTypes.func.isRequired
}

export default TodosItems
