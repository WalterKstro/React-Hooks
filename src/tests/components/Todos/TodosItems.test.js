import { shallow } from "enzyme"
import TodosItems from "../../../components/Todos/TodosItems"
import state from "../useReducer/dataTodos"

describe('Testing component <TodosItems/> in dir Todos', () => {

    let todo
    let handleToogle
    let handleDelete
    let component
    beforeEach(()=> {
        todo = state[0]
        handleToogle = jest.fn()
        handleDelete = jest.fn()
        component = shallow(<TodosItems todo={todo} handleDelete={handleDelete} handleToogle={handleToogle}/>)
    })

    test('should create a snapshots of component <TodosItems/>', () => {
        expect(component).toMatchSnapshot()
    })


    test('should call the function handleDelete', () => {
        component.find('button').simulate('click')
        expect(handleDelete).toHaveBeenCalledWith(todo.id)
    })

    test('should call the function handleToogle', () => {
        component.find('li').simulate('click')
        expect(handleToogle).toHaveBeenCalledWith(todo.id)
    })
    
    test('should show the content of tag <li>', () => {
        const contentTag = component.find('li').text()
        expect(contentTag).toBe(todo.task+'🗑️')
    })
    
    test('should has the clase text-decoration-line-through if the todo.state == true', () => {
        const todo = {id: 56884684698465,task: 'Aprender React',state: true}
        const component = shallow(<TodosItems todo={todo} handleDelete={handleDelete} handleToogle={handleToogle}/>)

        const {className} = component.find('li').props()
        expect(className.includes('text-decoration-line-through')).toBe(true)

    })
    
    
})
