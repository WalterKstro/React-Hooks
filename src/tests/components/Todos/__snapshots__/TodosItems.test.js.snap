// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <TodosItems/> in dir Todos should create a snapshots of component <TodosItems/> 1`] = `
<li
  className="false fw-bold list-group-item d-flex justify-content-between"
  key="56884684698465"
  onClick={[Function]}
>
  Aprender React
  <button
    className="btn btn-sm"
    onClick={[Function]}
  >
    🗑️
  </button>
</li>
`;
